### DISCUSSION TITLE
GPU frame rates and slow/frozen thresholds

### DISCUSSION THREAD

**Discussion by armcknight** [getsentry/sentry-cocoa#2595] [General]
Just took a look at the WWDC presentation for their Animation Hitches Instrument, where they describe some thresholds they use to determine bad animation states:

![image](https://user-images.githubusercontent.com/3241469/211855941-64d7ca00-c007-4853-8387-8ed52cff2cfb.png)

We do some calculations where we determine ["slow"](https://github.com/getsentry/sentry-cocoa/blob/27db30c11e9f1b897fe3d114e38720b9e8c63b36/Sources/Sentry/SentryFramesTracker.m#L114-L153) and "frozen" (700 ms) frames.

As we want to start overlaying slow/frozen frame renders on profile flamecharts, we are wondering how we can reconcile the threshold–and even naming/conceptual–differences between Apple's warning/critical and our slow/frozen. We want our tooling to be seamlessly transferrable to/from Apple's.

**Comment by armcknight** [comment]
@philipphofmann I am wondering where our [frozen frame threshold of 0.7](https://github.com/getsentry/sentry-cocoa/blob/27db30c11e9f1b897fe3d114e38720b9e8c63b36/Sources/Sentry/SentryFramesTracker.m#L18) (700ms) comes from? I see no mention of this decision in either https://github.com/getsentry/sentry-cocoa/pull/1123 or https://github.com/getsentry/sentry-cocoa/issues/1165

**Reply by philipphofmann** [reply]
Yep because of ⬇️ https://github.com/getsentry/sentry-cocoa/discussions/2595#discussioncomment-4668162 we choose these thresholds, but we could revisit these thresholds. I also have a proposal for some new measurements which I would call [frame delay](https://www.notion.so/sentry/Frame-Delay-0d6523df9d68423d8883986531f6dba1), which would give you the ms of delayed frames for a time period.

**Comment by stefanosiano** [comment]
For Android that threshold is recommended by Google in [Android Vitals].(https://developer.android.com/topic/performance/vitals/frozen)
It's also used by [Firebase](https://firebase.google.com/docs/perf-mon/screen-traces?platform=ios) in both Android and iOS sdks.
I'm not sure if Apple ever specified the "frozen frames" concept
